tests/test_lignin_kmc_parts.py

```diff
@@ -0,0 +1,874 @@
+GOOD_C_LIGNIN_MOL_OUT = os.path.join(SUB_DATA_DIR, 'c_lignin_molfile_good.txt')
+
+TCL_FNAME = "psfgen.tcl"
+TCL_FILE_LOC = os.path.join(SUB_DATA_DIR, TCL_FNAME)
+GOOD_TCL_OUT = os.path.join(SUB_DATA_DIR, "good_psfgen.tcl")
+GOOD_TCL_C_LIGNIN_OUT = os.path.join(SUB_DATA_DIR, "good_psfgen_c_lignin.tcl")
+GOOD_TCL_SHORT_SIM_OUT = os.path.join(SUB_DATA_DIR, "good_psfgen_short_sim.tcl")
+GOOD_TCL_NO_GROW_OUT = os.path.join(SUB_DATA_DIR, "good_psfgen_no_grow.tcl")
+GOOD_TCL_SHORT = os.path.join(SUB_DATA_DIR, "good_short_psfgen.tcl")
+
+# Data #
+SHORT_TIME = 0.00001
+ADJ_ZEROS = dok_matrix([[0, 0, 0, 0, 0],
+                        [0, 0, 0, 0, 0],
+                        [0, 0, 0, 0, 0],
+                        [0, 0, 0, 0, 0],
+                        [0, 0, 0, 0, 0]])
+
+MONO_DRAW_3 = [0.48772, 0.15174, 0.7886]
+MONO_DRAW_20 = [0.48772, 0.15174, 0.7886, 0.48772, 0.15174, 0.7886, 0.48772, 0.15174, 0.7886, 0.48772, 0.15174, 0.7886,
+                0.48772, 0.15174, 0.7886, 0.48772, 0.15174, 0.7886, 0.48772, 0.15174]
+
+ADJ2 = dok_matrix([[0, 4, 0, 0, 0],
+                   [8, 0, 0, 0, 0],
+                   [0, 0, 0, 8, 0],
+                   [0, 0, 5, 0, 0],
+                   [0, 0, 0, 0, 0]])
+
+ADJ3 = dok_matrix([[0, 4, 8, 0, 0],
+                   [8, 0, 0, 0, 0],
+                   [5, 0, 0, 0, 0],
+                   [0, 0, 0, 0, 0],
+                   [0, 0, 0, 0, 0]])
+
+
+def get_avg_num_bonds(bond_type, num_opts, result_list, num_repeats):
+    analysis = []
+    for i in range(num_opts):
+        opt_results = result_list[i]
+        cur_adjs = [opt_results[j][ADJ_MATRIX] for j in range(num_repeats)]
+        analysis.append([analyze_adj_matrix(cur_adjs[j]) for j in range(num_repeats)])
+
+    num_bonds = [[analysis[j][i][BONDS][bond_type]/sum(analysis[j][i][BONDS].values())
+                  for i in range(num_repeats)] for j in range(num_opts)]
+    av_num_bonds = [np.mean(bond_pcts) for bond_pcts in num_bonds]
+    std_num_bonds = [np.sqrt(np.var(bond_pcts)) for bond_pcts in num_bonds]
+    return av_num_bonds, std_num_bonds
+
+
+def create_sample_kmc_result(max_time=1., num_initial_monos=3, max_monos=10, sg_ratio=0.75, seed=10):
+    # The set lists are to minimize randomness in testing (adding while debugging source of randomness in some tests;
+    #     leaving because it doesn't hurt a thing; also leaving option to make a monomer_draw of arbitrary length
+    #     using a seed, but rounding those numbers because the machine precision differences in floats was the bug
+    np.random.seed(seed)
+    if num_initial_monos == 3:
+        monomer_draw = MONO_DRAW_3
+    elif num_initial_monos == 20:
+        monomer_draw = [0.77132064, 0.02075195, 0.63364823, 0.74880388, 0.49850701, 0.22479665, 0.19806286,
+                        0.76053071, 0.16911084, 0.08833981, 0.68535982, 0.95339335, 0.00394827, 0.51219226,
+                        0.81262096, 0.61252607, 0.72175532, 0.29187607, 0.91777412, 0.71457578]
```

Comment:
Was this different than MONOMER_DRAW_20 for any reason? (Recognizing that it won't actually impact how it works)

Reply:
there are two options I explicitly tested--starting with 3 and 20 monomers. For those, I made my set distribution, to reduce random calls. I left the option to make this more general by allowing other options